lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -55,12 +55,65 @@ const SessionAuth: React.FC<PropsWithChildren<Props>> = ({ children, ...props })
     const parentSessionContext = useContext(SessionContext);
 
     // assign the parent context here itself so that there is no flicker in the UI
-    const [context, setContext] = useState<SessionContextType | undefined>(
-        hasParentProvider(parentSessionContext) ? parentSessionContext : undefined
-    );
+    const [context, setContext] = useState<SessionContextType | undefined>(undefined);
 
+    const [contextUpdate, setContextUpdate] = useState<SessionContextTypeWithoutInvalidClaim | undefined>(undefined);
     const session = useRef(Session.getInstanceOrThrow());
 
+    useEffect(() => {
+        const abortController = new AbortController();
+        async function effect() {
+            let invalidClaim: ClaimValidationError | undefined = undefined;
+            if (contextUpdate === undefined) {
+                return;
+            }
+            if (contextUpdate.doesSessionExist && props.overwriteDefaultClaimValidators !== undefined) {
+                const userContext = {};
+                // TODO: user proper userContext
+                // TODO: only one of these should be running at any time, but refreshing a claim could start this again.
+                let accessTokenPayload = contextUpdate.accessTokenPayload;
+                const defaultValidators = session.current.getDefaultClaimValidators();
+
+                const requiredClaims =
+                    props.overwriteDefaultClaimValidators !== undefined
+                        ? props.overwriteDefaultClaimValidators(defaultValidators)
+                        : defaultValidators;
+
+                // We first refresh all claims that needs this, so we avoid ha
+                for (const validator of requiredClaims) {
+                    if (await validator.shouldRefresh(accessTokenPayload, userContext)) {
+                        await validator.refresh(userContext);
+                        accessTokenPayload = await session.current.getAccessTokenPayloadSecurely();
+                    }
+                    const validationRes = await validator.validate(accessTokenPayload, userContext);
+                    if (!validationRes.isValid) {
+                        invalidClaim = {
+                            validatorId: validator.id,
+                            reason: validationRes.reason,
+                        };
```

Comment:
@porcellus missed this?

Reply:
No, but we've discussed this on a call before, and I thought we'd address it in a separate PR with the email verification claims. So basically there are a few options:
1. Special casing this if it's added by the other recipe. This can be done by adding a version of the claim as a global claim that can tell the session verification recipe to redirect. The `SuperTokensWrapper` will have to disable this to avoid a redirection loop.
3. Adding specific config into the session auth wrapper to do these kinds of redirections (or do them if `requireAuth` is set to true)

Reply:
Keeping this open for email veirifcation PR